src/error-once.md

```diff
@@ -0,0 +1,112 @@
+# Handle Errors Once
+
+When a function receives an error from a caller,
+it can handle it in a variety of different ways
+depending on what it knows about the error.
+
+These include, but not are limited to:
+
+- if the function contract defines specific errors,
```

Comment:
nit: function can refer to the caller or callee, what do you think of explicitly using those terms throughout? "if the callee function defines specific errors"

Reply:
oh good call. will change.